torch_geometric/nn/conv/point_transformer_conv.py

```diff
@@ -0,0 +1,110 @@
+from torch_geometric.typing import OptTensor, PairOptTensor, PairTensor, Adj
+
+from torch import Tensor
+from torch_sparse import SparseTensor, set_diag
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.utils import remove_self_loops, add_self_loops
+
+from torch.nn import Sequential, ReLU, Linear
+from torch.nn.functional import softmax
+
+from ..inits import reset
+
+
+class PointTransformerConv(MessagePassing):
+    r"""The PointTransformer layer from the `"Point Transformer"
+    <https://arxiv.org/abs/2012.09164>` paper
+
+    .. math::
+        \mathbf{x}^{\prime}_i =  \sum_{j \in
+        \mathcal{N}(i) \cup \{ i \}} \rho \left( \gamma \left( \phi
+        \left( \mathbf{x}_i \right) - \psi \left( \mathbf{x}_j \right) \right)
+        + \delta  \right) \odot \left( \alpha \left(  \mathbf{x}_j \right)
+        + \delta \right) ,
+
+    where :math:`\gamma` and :math:`\delta = \theta \left( \mathbf{p}_j -
+    \mathbf{p}_i \right)` denote neural networks, *.i.e.* MLPs, that
+    respectively map transformed node features and relative spatial
+    coordinates. :math:`\mathbf{P} \in \mathbb{R}^{N \times D}`
+    defines the position of each point.
+
+    Args:
+        in_channels (int): Size of each input sample features
+        out_channels (int) : size of each output sample features
+        add_self_loops (bool, optional): If set to :obj:`False`, will not add
+            self-loops to the input graph. (default: :obj:`True`)
+
+        pos_mlp_channels : (int, optional) size of intermediary channels in
+            points relative spatial coordinates computation.
+            (default: :obj:`64`)
+        attn_mlp_channels : (int, optional) factor size of intermediary
+            channels in transformed features mlp. (default: :obj:`2`)
+
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
+    """
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 add_self_loops=True,
+                 pos_mlp_channels=64,
+                 attn_mlp_channels=2):
+        super(PointTransformerConv,
+              self).__init__(aggr='add')  # "Add" aggregation
+
+        self.delta = Sequential(Linear(3, pos_mlp_channels), ReLU(),
+                                Linear(pos_mlp_channels, out_channels))
+        self.gamma = Sequential(
+            Linear(out_channels, out_channels * attn_mlp_channels), ReLU(),
+            Linear(out_channels * attn_mlp_channels, out_channels))
+        self.alpha = Linear(in_channels, out_channels)
+        self.phi = Linear(in_channels, out_channels)
+        self.psi = Linear(in_channels, out_channels)
+
+        self.add_self_loops = add_self_loops
+
+    def reset_parameters(self):
+        reset(self.delta)
+        reset(self.gamma)
+        reset(self.alpha)
+        reset(self.phi)
+        reset(self.psi)
+
+    def forward(self, x: Union[OptTensor, PairOptTensor],
```

Comment:
To be honest, I didn't fully grasp the architecture of functions, this is why I copy pasted this part from PointConv that I used as a reference. As rusty1 pointed out in this discussion https://github.com/pyg-team/pytorch_geometric/discussions/3169 we can possibly work without features with the `ToConstant(1)` transform trick. 
However I personaly think it would be better to make it the user's responsability however, so I agree with you